src/components/DatabaseSearch/DatabaseSearchSlice.js

```diff
@@ -27,7 +27,13 @@ const slice = createSlice({
       })
       .addCase(searchByName.fulfilled, (state, action) => {
         state.status = 'idle';
-        state.searchResults = action.payload;
+        if(!state.searchResults.map(result => result.id).includes(action.payload.id)) {
+          state.searchResults.push(action.payload)
+        }
+      })
+      .addCase(searchByName.rejected, (state) => {
+        state.status = 'idle';
+        state.searchResults = []
       })
```

Comment:
Search functionality is far from ideal - given more time I would ask backend to send this information back in an array. That would make it much easier to work with. Also not sure by the search endpoint is only returning one result for each query. Seems unlikely.